### PR TITLE
362 Update deterministic settings on several bundles

### LINKED ISSUE #362: Update bundles to enable deterministic training
I did some experiments on some bundles to see if they can support deterministic training.

## Methods
The method I used is to run training (both single gpu and multi gpus) twice for each bundle, and compare the achieved weights from different runs. I added a variable "output_name" in `train.json` config and it was used in the `CheckpointSaver` handler. For bundles that can be trained fast, I trained a few epochs and for bundles that have large datasets, I set `epoch_length` to a small number in `trainer` and `evaluator` to reduce the test time.

The training command is like:
```
torchrun --standalone --nnodes=1 --nproc_per_node=2 -m monai.bundle run --config_file "['configs/train.json','configs/multi_gpu_train.json']" --dataset_dir <your-dataset>
torchrun --standalone --nnodes=1 --nproc_per_node=2 -m monai.bundle run --config_file "['configs/train.json','configs/multi_gpu_train.json']" --dataset_dir <your-dataset> --output_name model_2.pt
```

After training, I use the code like the following to compare achieved weights:

```
import torch
import os
# print(os.listdir())

bundle = "lung_nodule_ct_detection"

m1 = torch.load(os.path.join(bundle, "models/model.pt"))
m2 = torch.load(os.path.join(bundle, "models/model_2.pt"))

for k in m1.keys():
    if not torch.all(m1[k] == m2[k]):
        print(k)
```

## Results

Bundles that shown in black font color support deterministic training:

Screen Shot 2023-04-14 at 4 14 50 PM

## Plan of changes

I will change bundles to enable deterministic training, and for unsupported bundles I will mention it in the readme file. All changes limit to bundles shown in the above table.

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes #362 .

### Description
This PR adds deterministic settings on several bundles mentioned in #362 , and for unsupported bundles, I added some notes on readme files.

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [x] In-line docstrings updated.
- [x] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [x] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Review by Nic-Ma — APPROVED** [review]

File: models/pathology_nuclei_segmentation_classification/docs/README.md [hunk 1]
```diff
@@ -1,21 +1,18 @@
 # Model Overview
-
 A pre-trained model for simultaneous segmentation and classification of nuclei within multi-tissue histology images based on CoNSeP data. The details of the model can be found in [1].
 
 ## Workflow
+The model is trained to simultaneous segment and classify nuclei. Training is done via a two-stage approach. First initialized the model with pre-trained weights on the [ImageNet dataset](https://ieeexplore.ieee.org/document/5206848), trained only the decoders for the first 50 epochs, and then fine-tuned all layers for another 50 epochs. There are two training modes in total. If "original" mode is specified, [270, 270] and [80, 80] are used for `patch_size` and `out_size` respectively. If "fast" mode is specified, [256, 256] and [164, 164] are used for `patch_size` and `out_size` respectively. The results shown below are based on the "fast" mode.
```
Comment (KumoLiu) [inline models/pathology_nuclei_segmentation_classification/docs/README.md]:
```suggestion
The model is trained to simultaneously segment and classify nuclei. Training is done via a two-stage approach. First initialized the model with pre-trained weights on the [ImageNet dataset](https://ieeexplore.ieee.org/document/5206848), trained only the decoders for the first 50 epochs, and then fine-tuned all layers for another 50 epochs. There are two training modes in total. If "original" mode is specified, [270, 270] and [80, 80] are used for `patch_size` and `out_size` respectively. If "fast" mode is specified, [256, 256] and [164, 164] are used for `patch_size` and `out_size` respectively. The results shown below are based on the "fast" mode.
```

**Review by KumoLiu — APPROVED** [review]
Thanks for update, looks good to me.

**Pull request merged**